### DISCUSSION TITLE
urequests in ver 1.20

### DISCUSSION THREAD

**Discussion by kjm1102** [micropython/micropython#12077] [Libraries & Drivers]
I wanted to take a look at the code in the latest urequests but can't figure out how to view it? If I dir(urequests) I see a list of supported cmds but I really need to see what changes have been made since the version (a modified circa ver 1.17 fiddled to allow socket timeouts) I'm currently using.

**Comment by jimmo** [comment]
`urequests` was just renamed to `requests` yesterday. (However `import urequests` will continue to work).

The code is at https://github.com/micropython/micropython-lib/blob/master/python-ecosys/requests/requests/__init__.py

(And the `urequests` wrapper is here https://github.com/micropython/micropython-lib/blob/master/micropython/urequests/urequests.py)

Until yesterday, it was here https://github.com/micropython/micropython-lib/blob/v1.20.0/python-ecosys/urequests/urequests.py  (this is the v1.20.0 version). It gets added to the firmware via the micropython-lib submodule.

FYI - almost all python dependencies are brought in from micropython-lib.

**Comment by kjm1102** [comment]
Tnx Jim. Is there a way to tell if settimeout is likely to work on a particular board other than buying one & trying it?

**Reply by jimmo** [reply]
settimeout is supported on all LWIP-based boards (ESP32, ESP8266, Pico W, Pyboard D, WizNet in MAC-RAW mode). Some of the non-LWIP boards also support it (e.g. Arduino Nano Connect) a well as the unix port, but not WizNet boards where the chip runs the TCP stack.

(In other words, most things do support it)

**Reply by kjm1102** [reply]
Tnx Jim. Calling the new improved urequests in ver 1.20 with a timeout so it doesn't waste battery power waiting an eternity on ratty connections for the likes of [Errno 118] EHOSTUNREACH or [Errno 113] ECONNABORTED or any of the others from the litany of comms errors seems to work but I'm intrigued by the response
```  
File "urequests.py", line 125, in request
OSError: -116  
```
is that because the socket has timed out? So I can expect OSError: -116 consistently for socket timeouts?

**Reply by jimmo** [reply]
The negative number implies that this is an error from a third-party library (e.g. lwip or mbedtls) that couldn't be mapped into a standard `EFOO` code, but I don't see where 116 is coming from. Which port are you using? ESP32?

**Reply by kjm1102** [reply]
Using V1.20 for esp32 Jim, I'm pretty sure the -116 is when the socket in urequests times out
.
